src/app/state/account/saga.ts

```diff
@@ -130,6 +131,8 @@ export function* fetchingOnAccountPage() {
             staleBalances.delegations !== refreshedAccount.delegations ||
             staleBalances.debonding !== refreshedAccount.debonding
           ) {
+            // Wait for oasisscan to update transactions (it updates balances faster)
+            yield* delay(TRANSACTIONS_UPDATE_DELAY)
```

Comment:
Even though, thinking about this, this may introduce race condition. In a sense that, if you switch account in the mean time while the delay is in progress. It could refresh the "newly" opened account for no reason. So it should check `state.address !== accountAddress`(should be provided from payload action). Will try to work it out in https://github.com/oasisprotocol/oasis-wallet-web/pull/1954

Reply:
can you start with a test for this? I don't think it happens

Reply:
Yeah, you are right `delay` gets canceled by `takeLatestCancelable`. Never mind. I though it acts as a setTimeout(without it being canceled).